### DISCUSSION TITLE
Sending CSRF Token on POST request laravel

### DISCUSSION THREAD

**Discussion by hairullana** [thunderclient/thunder-client-support#885] [General]
As you know that in laravel when we need to send a request with POST method, we need to send csrf token that we can generate it with @csrf() on blade, but I little confusing when we want to test some api especially with thunder client, how we can get csrf token and send to laravel?

**Comment by rangav** [comment]
Hi you can save csrf token to env variable  from response and use it in the following requests 

https://github.com/rangav/thunder-client-support#setenv

**Comment by sadiqsalau** [comment]
1. Create an environment.
2. Add a test to your collection by navigating to the `Tests` tab, then select `Set Env Variable`, set query to `cookie.xsrf-token` and value to `{{XSRF-TOKEN}}`
3. In the `Headers` tab, add an header named `X-XSRF-TOKEN` with value `{{XSRF-TOKEN | urlDecode}}`.
4. Create a request to `/sanctum/csrf-cookie` to refresh CSRF Token if it expires

**Reply by rangav** [reply]
Thanks @sadiqsalau for sharing steps, will add to the documentation
